WebRoot/js/zimbraMail/prefs/view/ZmSharingPage.js

```diff
@@ -835,11 +891,37 @@ function(html, idx, item, field, colIdx, params) {
 		} else {
 			idx = this._addActionLinks(item, html, idx);
 		}
+	} else if (field === ZmSharingView.F_SYNC) {
+		var disableSyncToMobile = [ZmShare.ROLE_ADMIN, ZmShare.ROLE_MANAGER].indexOf(item.link.role) === -1 ||
+									["appointment", "message", "contact"].indexOf(item.link.view) === -1;
+		var disabledAttribute = disableSyncToMobile ? 'disabled' : '';
+		var checkedAttribute = item.mountpoint.activeSyncDisabled === false ? 'checked' : ''; 
+		var labelId = this.getHeaderId(ZmSharingView.F_SYNC);
+		html[idx++] = "<input type='checkbox' "+ checkedAttribute +" id='"+ item.mountpoint.id +"' aria-labelledby='"+ labelId +"' onchange='ZmSharingView._handleMobileSync(this)' "+ disabledAttribute +" />"
 	}
 
 	return (params && params.returnText) ? html.join("") : idx;
 };
 
+ZmSharingListView.prototype._emulateSingleClick =
+function(params) {
+	if (
+		this.status === ZmSharingView.MOUNTED &&
+		params.ctrlKey &&
```

Comment:
Just a question. What value is expected in `params.ctrlKey`? boolean?
Is this param related to the following comment? If so, how is backquote considered?
```
handle selection of mobile sync checkbox using keyboard shortcut Ctrl + `.
```

Reply:
The value of `params.ctrlKey` is boolean. it will be true when user select the list item using keyboard shortcut  **Ctrl + backtick**.

The `_emulateSingleClick` is going to be called when user navigate to next/prev line using keyboard "Up/Down" arrow keys and selecting the list item using **Ctrl + backtick**.  

here we need tick/untick Sync to mobile checkbox when user perform **Ctrl + backtick** only hence we are just checking with params.ctrlKey value.

Reply:
I understand. Thanks!